File: test_tickets.js
const axios = require('axios');

// Configuration
const API_BASE = 'http://localhost:8002';
const STORE_ID = '1539874562';
const TODAY_DATE = '20260319'; // YYYYMMDD format

// Helper to create a ticket object
function createTicket(ticketNum, closureNumber) {
  return {
    IdCRM: STORE_ID,
    Date: TODAY_DATE,
    idTiquer: ticketNum.toString(),
    HeureTicket: `${String(8 + Math.floor(ticketNum / 3)).padStart(2, '0')}:${String((ticketNum % 60) * 2).padStart(2, '0')}`,
    TTC: Math.round((20 + Math.random() * 80) * 100) / 100,
    currency: '€',
    merchant_name: 'TEST RESTAURANT',
    merchant_address: '123 TEST STREET',
    SIRET: '12345678901234',
    Z: closureNumber, // Closure number / shift
    ConsumptionMode: ['Dine-in', 'Takeaway', 'Delivery'][ticketNum % 3],
    Menu: [
      {
        NameProduct: `MENU ${ticketNum}`,
        TTC: Math.round((15 + Math.random() * 50) * 100) / 100,
        QtyProduct: 1,
      }
    ],
    PaymentMethods: [
      {
        payment_method: ['CASH', 'CARD', 'CHECK'][ticketNum % 3],
        amount: Math.round((20 + Math.random() * 80) * 100) / 100,
      }
    ],
    Totals: {
      Total_Ht: Math.round((15 + Math.random() * 70) * 100) / 100,
      Total_TVA: Math.round((3 + Math.random() * 10) * 100) / 100,
      Total_TTC: Math.round((20 + Math.random() * 80) * 100) / 100,
    },
  };
}

// Main function
async function postTickets() {
  console.log('🎫 Starting ticket posting test...\n');
  console.log(`Store ID: ${STORE_ID}`);
  console.log(`Date: ${TODAY_DATE}`);
  console.log(`Creating 20 tickets with closure numbers 0-19\n`);

  const closureNumbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0];

  let successCount = 0;
  let failureCount = 0;
  const results = [];

  for (let i = 1; i <= 20; i++) {
    const closureNum = closureNumbers[i - 1];
    const ticket = createTicket(i, closureNum);

    try {
      console.log(`[${i}/20] Posting ticket ${i} (Closure: ${closureNum})...`);
      
      const response = await axios.post(`${API_BASE}/update-ticket`, ticket, {
        headers: {
          'Content-Type': 'application/json',
        },
        timeout: 10000,
      });

      console.log(`  ✅ Success - ID: ${response.data._id || 'N/A'}`);
      successCount++;
      results.push({
        ticketNum: i,
        closure: closureNum,
        status: 'success',
        amount: ticket.TTC,
      });
    } catch (error) {
      console.log(`  ❌ Failed - ${error.response?.data?.error || error.message}`);
      failureCount++;
      results.push({
        ticketNum: i,
        closure: closureNum,
        status: 'failed',
        error: error.response?.data?.error || error.message,
      });
    }

    // Small delay between requests
    await new Promise(resolve => setTimeout(resolve, 300));
  }

  // Summary
  console.log(`\n${'='.repeat(80)}`);
  console.log('📊 TEST SUMMARY');
  console.log(`${'='.repeat(80)}`);
  console.log(`✅ Successful: ${successCount}/20`);
  console.log(`❌ Failed: ${failureCount}/20`);

  // Group by closure number
  const byShift = {};
  results.forEach(r => {
    if (!byShift[r.closure]) byShift[r.closure] = [];
    byShift[r.closure].push(r);
  });

  console.log(`\n📋 Breakdown by Shift:`);
  Object.keys(byShift).sort((a, b) => a - b).forEach(shift => {
    const shiftTickets = byShift[shift];
    const successInShift = shiftTickets.filter(t => t.status === 'success').length;
    console.log(`  Shift ${shift}: ${successInShift}/${shiftTickets.length} tickets`);
  });

  console.log(`\n${'='.repeat(80)}`);
  console.log('Test Complete! 🎉\n');
  console.log('Next Steps:');
  console.log('1. Navigate to Dashboard to see the new tickets');
  console.log('2. Use "Filter by Shift" dropdown to filter by closure number');
  console.log('3. Click "Shifts" in sidebar to see shift summary page');
  console.log('4. Check API: GET /get-tickets?idCRM=1539874562&date1=20260319&date2=20260319&closureNumber=0');
}

postTickets().catch(error => {
  console.error('Fatal error:', error);
  process.exit(1);
});

File: test_tickets.py
#!/usr/bin/env python3
import subprocess
import json
from datetime import datetime

API_BASE = "http://localhost:8002"
STORE_ID = "1539874562"
TODAY_DATE = "20260319"

def create_ticket(ticket_num, closure):
    """Create a ticket payload"""
    hour = 8 + (ticket_num // 3)
    minute = (ticket_num % 60) * 2
    amount = 20 + (ticket_num * 3.5) % 80
    
    return {
        "IdCRM": STORE_ID,
        "Date": TODAY_DATE,
        "idTiquer": str(ticket_num),
        "HeureTicket": f"{hour:02d}:{minute:02d}",
        "TTC": round(amount, 2),
        "currency": "€",
        "merchant_name": "TEST RESTAURANT",
        "merchant_address": "123 TEST STREET",
        "SIRET": "12345678901234",
        "Z": closure,  # Closure number / shift
        "ConsumptionMode": ["Dine-in", "Takeaway", "Delivery"][ticket_num % 3],
        "Menu": [
            {
                "NameProduct": f"MENU {ticket_num}",
                "TTC": 25.00,
                "QtyProduct": 1
            }
        ],
        "PaymentMethods": [
            {
                "payment_method": ["CASH", "CARD", "CHECK"][ticket_num % 3],
                "amount": round(amount, 2)
            }
        ],
        "Totals": {
            "Total_Ht": 20.00,
            "Total_TVA": 5.00,
            "Total_TTC": round(amount, 2)
        }
    }

def main():
    print("🎫 Starting ticket posting test...")
    print(f"Store ID: {STORE_ID}")
    print(f"Date: {TODAY_DATE}")
    print("Creating 20 tickets with closure numbers 0-19\n")
    
    # Closure numbers: 0 (unassigned), then 1-9 repeated
    closure_numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0]
    
    success_count = 0
    failed_count = 0
    
    for i in range(1, 21):
        closure = closure_numbers[i - 1]
        ticket = create_ticket(i, closure)
        
        print(f"[{i}/20] Posting ticket {i} (Closure: {closure})...", end=" ", flush=True)
        
        try:
            # Use curl to post
            cmd = [
                "curl",
                "-s",
                "-X", "POST",
                f"{API_BASE}/update-ticket",
                "-H", "Content-Type: application/json",
                "-d", json.dumps(ticket)
            ]
            
            result = subprocess.run(cmd, capture_output=True, text=True, timeout=10)
            
            # Check if response contains success indicators
            if result.returncode == 0:
                response = result.stdout
                if "_id" in response or "successfully" in response.lower():
                    print("✅ Success")
                    success_count += 1
                else:
                    print(f"❌ Failed - {response[:100]}")
                    failed_count += 1
            else:
                print(f"❌ Failed - {result.stderr[:100]}")
                failed_count += 1
        except Exception as e:
            print(f"❌ Error - {str(e)[:50]}")
            failed_count += 1
    
    # Summary
    print(f"\n{'='*80}")
    print("📊 TEST SUMMARY")
    print(f"{'='*80}")
    print(f"✅ Successful: {success_count}/20")
    print(f"❌ Failed: {failed_count}/20")
    print(f"\n{'='*80}")
    print("Next Steps:")
    print("1. Navigate to Dashboard to see the new tickets")
    print("2. Use 'Filter by Shift' dropdown to filter by closure number")
    print("3. Click 'Shifts' in sidebar to see shift summary page")
    print("4. Test API filtering:")
    print(f"   curl 'http://localhost:8002/get-tickets?idCRM={STORE_ID}&date1={TODAY_DATE}&date2={TODAY_DATE}&closureNumber=0'")
    print(f"\n")

if __name__ == "__main__":
    main()
